packages/vitest/src/node/cache/fsCache.ts

```diff
@@ -0,0 +1,270 @@
+      return
+    }
+
+    const meta = this.fromBase64(code.slice(matchIndex + cacheCommentLength))
+    if (meta.externalize) {
+      debugFs?.(`${c.green('[read]')} ${meta.externalize} is externalized inside ${cachedFilePath}`)
+      return { externalize: meta.externalize, type: meta.type }
+    }
+    debugFs?.(`${c.green('[read]')} ${meta.id} is cached in ${cachedFilePath}`)
+
+    return {
+      id: meta.id,
+      url: meta.url,
+      file: meta.file,
+      code,
+      importers: meta.importers,
+      mappings: meta.mappings,
+    }
+  }
+
+  async saveCachedModule<T extends FetchResult>(
+    cachedFilePath: string,
+    fetchResult: T,
+    importers: string[] = [],
+    mappings: boolean = false,
+  ): Promise<void> {
+    if ('externalize' in fetchResult) {
+      debugFs?.(`${c.yellow('[write]')} ${fetchResult.externalize} is externalized inside ${cachedFilePath}`)
+      await atomicWriteFile(cachedFilePath, `${cacheComment}${this.toBase64(fetchResult)}`)
+    }
+    else if ('code' in fetchResult) {
+      const result = {
+        file: fetchResult.file,
+        id: fetchResult.id,
+        url: fetchResult.url,
+        importers,
+        mappings,
+      } satisfies Omit<FetchResult, 'code' | 'invalidate'>
+      debugFs?.(`${c.yellow('[write]')} ${fetchResult.id} is cached in ${cachedFilePath}`)
+      await atomicWriteFile(cachedFilePath, `${fetchResult.code}${cacheComment}${this.toBase64(result)}`)
+    }
+  }
+
+  private toBase64(obj: unknown) {
+    const json = stringify(obj)
+    return Buffer.from(json).toString('base64')
+  }
+
+  private fromBase64(obj: string) {
+    const json = Buffer.from(obj, 'base64').toString('utf-8')
+    return parse(json)
+  }
+
+  invalidateCachePath(
+    environment: DevEnvironment,
+    id: string,
+  ): void {
+    debugFs?.(`cache for ${id} in ${environment.name} environment is invalidated`)
+    this.fsCacheKeys.get(environment)?.delete(id)
+  }
+
+  invalidateAllCachePaths(environment: DevEnvironment): void {
+    debugFs?.(`the ${environment.name} environment cache is invalidated`)
+    this.fsCacheKeys.get(environment)?.clear()
+  }
+
+  getMemoryCachePath(
+    environment: DevEnvironment,
+    id: string,
+  ): string | undefined {
+    const result = this.fsCacheKeys.get(environment)?.get(id)
+    if (result) {
+      debugMemory?.(`${c.green('[write]')} ${id} was cached in ${result}`)
+    }
+    return result
+  }
+
+  generateCachePath(
+    vitestConfig: ResolvedConfig,
+    environment: DevEnvironment,
+    resolver: VitestResolver,
+    id: string,
+    fileContent: string,
+  ): string {
+    const config = environment.config
+    // coverage provider is dynamic, so we also clear the whole cache if
+    // vitest.enableCoverage/vitest.disableCoverage is called
+    const coverageAffectsCache = !!(this.vitest.config.coverage.enabled && this.vitest.coverageProvider?.requiresTransform?.(id))
+    const cacheConfig = JSON.stringify(
```

Comment:
Wouldn't it benefit to reuse this config-wide cache-key instead of recomputing for each `id/fileContent`?